packages/react/package.json

```diff
@@ -46,5 +46,9 @@
 	"peerDependencies": {
 		"@tanstack/react-query": "^5",
 		"firebase": "^11.3.0"
+	},
+	"dependencies": {
+		"@types/deep-equal": "^1.0.4",
+		"deep-equal": "^2.2.3"
```

Comment:
I'm a bit hesitant to pull in a dependency for this vs. just writing a simple recursive comparison function...

Reply:
Yeah I was literally writing out the full function when I thought your feedback would be the opposite. I can just write the comparison function

Reply:
The best number of dependencies is zero. The next best number is "as few high-quality and well-maintained dependencies as possible that save weeks+ of effort or ongoing maintenance".

Reply:
Do we not already have some kind of ref comparison function in the core SDK? That's kinda where I'd expect this to be implemented.

Reply:
We don't. I'll implement it here first where we need it, but I'll make sure we get something in the next version of the JS SDK

Reply:
Wrote the deep equals function